sdk/core/core-http/src/serializer.ts

```diff
@@ -918,6 +918,7 @@ function deserializeCompositeType(
             propertyObjectName,
             options
           );
+          handledPropertyNames.push(xmlName!);
```

Comment:
I thought about whether we should care about marked seralizedName earlier at `handledPropertyNames.push(paths[0]);`.  If the two names are the same, then we are marking the same property so that's fine.  If the two are different, I wouldn't expect the parsed JSON has a property with `seralizedName` unless people intentionally introduce a name conflict with other properties, i.e., serializedName is same as other XML names.  In this case, marking it doesn't matter either, as the proeprty will be marked too by their xml name.